src/modules/ekf2/EKF/aid_sources/fake_pos_control.cpp

```diff
@@ -78,7 +78,7 @@ void Ekf::controlFakePosFusion()
 		const bool enable_valid_fake_pos = _control_status.flags.constant_pos || _control_status.flags.vehicle_at_rest;
 		const bool enable_fake_pos = !enable_valid_fake_pos
 					     && (getTiltVariance() > sq(math::radians(3.f)))
-					     && !(_params.imu_ctrl & static_cast<int32_t>(ImuCtrl::GravityVector))
+					     && !_control_status.flags.gravity_vector
```

Comment:
The gravity_vector flag is handled a bit differently where it might be flipping on/off relatively rapidly with the accel threshold. Should we change that behavior so it's more consistent with other aid sources? Alternatively we could check here if gravity fusion hasn't been active for some period of time before finally giving up and starting fake_pos.

https://github.com/PX4/PX4-Autopilot/blob/629f460e1f9a39f055c5daa7283a55022cb44c7b/src/modules/ekf2/EKF/aid_sources/gravity/gravity_fusion.cpp#L62-L64

Reply:
Sorry, completely missed the plot in the PR description about the rapid switching. 

Are we completely sure additional protection isn't needed so that we only start fake_pos as a last resort given it will reset pos/vel on start?

Reply:
The tilt variance condition `(getTiltVariance() > sq(math::radians(3.f)))` should make sure the fake position doesn't start immediately when the gravity fusion was healthy and briefly switches off (it should take several seconds to reach 3 degrees of tilt uncertainty).